controllers/replica_set_controller.go

```diff
@@ -559,6 +562,29 @@ func buildAutomationConfig(mdb mdbv1.MongoDBCommunity, auth automationconfig.Aut
 		Build()
 }
 
+func guessEnterprise(mdb mdbv1.MongoDBCommunity) bool {
+	overrideAssumption, err := strconv.ParseBool(os.Getenv(construct.MongoDBAssumeEnterpriseEnv))
+	if err == nil {
+		return overrideAssumption
+	}
+
+	var overriddenImage string
+	containers := mdb.Spec.StatefulSetConfiguration.SpecWrapper.Spec.Template.Spec.Containers
+	if len(containers) > 0 {
+		for _, c := range containers {
+			if c.Name == construct.MongodbName {
+				if len(c.Image) > 0 {
+					overriddenImage = c.Image
+				}
+			}
+		}
+	}
+	if strings.Contains(overriddenImage, construct.OfficialMongodbEnterpriseServerImageName) {
```

Comment:
shouldn't we return false if we have an override which is **not** enterprise? Imagine the testcase which I mention further below

Reply:
Answering this here https://github.com/mongodb/mongodb-kubernetes-operator/pull/1271#discussion_r1171775852